File: src/main/java/com/hsmchurch/app/reply/domain/ReplyRepository.java
package com.hsmchurch.app.reply.domain;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface ReplyRepository extends JpaRepository<Reply, Long>, ReplyRepositoryCustom {

    Page<Reply> findAllByReplier_Id(Long writerId, Pageable pageable);

    List<Reply> findAllByVideoIdAndReplier_Id(Long videoId, Long replierId);

    List<Reply> findAllByVideoId(Long videoId);

}

File: src/main/java/com/hsmchurch/app/reply/domain/ReplyRepository.kt
package com.hsmchurch.app.reply.domain

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface ReplyRepository : JpaRepository<Reply, Long>, ReplyRepositoryCustom {

    fun findAllByReplier_Id(writerId: Long, pageable: Pageable): Page<Reply>

    fun findAllByVideoIdAndReplier_Id(videoId: Long, replierId: Long): List<Reply>

}
